### PR TITLE
Enable driver jar publish

### PR DESCRIPTION
**Pull request by ashetkar**
- Enable driver jar publish
- Added missing gradle config post-upgrade

### CODE REVIEW

File: build-logic/publishing/src/main/kotlin/build-logic.java-published-library.gradle.kts [hunk 1]
```diff
@@ -17,6 +17,8 @@ java {
 publishing {
     publications {
         create<MavenPublication>(project.name) {
+            artifactId = "jdbc-yugabytedb"
```
Comment (kneeraj) [inline build-logic/publishing/src/main/kotlin/build-logic.java-published-library.gradle.kts:20]: Why does the pgjdbc upstream code does not need to specify the artifactId and version here but we need to?

File: build-logic/publishing/src/main/kotlin/build-logic.publish-to-tmp-maven-repo.gradle.kts [hunk 2]
```diff
@@ -22,6 +22,14 @@ publishing {
             name = "tmp-maven"
             url = uri(localRepoDir)
         }
+        maven {
+            name = "remote"
+            url = uri("https://oss.sonatype.org/service/local/staging/deploy/maven2/")
+            credentials {
+                username = System.getenv("OSSRH_USERNAME")
```
Comment (kneeraj) [inline build-logic/publishing/src/main/kotlin/build-logic.publish-to-tmp-maven-repo.gradle.kts:29]: Similar to the first comment - How is pgjdbc taking the credentials from the environment?

**Review by kneeraj — COMMENTED** [review]
Two questions to understand our modification's need. Rest look ok.

File: build-logic/publishing/src/main/kotlin/build-logic.java-published-library.gradle.kts (hunk 1 above)
Comment (ashetkar) [reply build-logic/publishing/src/main/kotlin/build-logic.java-published-library.gradle.kts:20]: I had ported all the changes from previous master that were missing.
But now if I remove only this change, I'm still able to publish the jar. May be we can get rid of this then.

File: build-logic/publishing/src/main/kotlin/build-logic.publish-to-tmp-maven-repo.gradle.kts (hunk 2 above)
Comment (ashetkar) [reply build-logic/publishing/src/main/kotlin/build-logic.publish-to-tmp-maven-repo.gradle.kts:29]: Do not know for sure, but there are additional config files ignored in `.gitignore` which may be providing such details. These are `release.properties`, `*.local.properties`, `pom.xml.*`, `dependency-reduced-pom.xml`.
But then, I do not see these being referred anywhere in the repository either.

File: build-logic/publishing/src/main/kotlin/build-logic.java-published-library.gradle.kts (hunk 1 above)
Comment (ashetkar) [reply build-logic/publishing/src/main/kotlin/build-logic.java-published-library.gradle.kts]: Reverted this change

File: build-logic/publishing/src/main/kotlin/build-logic.publish-to-tmp-maven-repo.gradle.kts (hunk 2 above)
Comment (kneeraj) [reply build-logic/publishing/src/main/kotlin/build-logic.publish-to-tmp-maven-repo.gradle.kts:29]: Let's find this out. Not blocking the PR for this but let's understand this.

**Review by kneeraj — APPROVED** [review]

**Comment by ashetkar** [comment]
With this change, the gradle task to publish the jar changes from
```
./gradlew publishPostgresqlPublicationToRemoteRepository -x test -Prelease
```
to
```
./gradlew publishJdbc-yugabytedbPublicationToRemoteRepository -x test -Prelease
```

**Pull request merged**